### ISSUE TITLE
error: unknown command with a -e flag (bun)

### ISSUE THREAD

**Issue by kingmesal** [tj/commander.js#2205]
I had the following code for version 9.4.1. When I upgraded to 12.1.0 this no longer works. 

When I execute `cli.ts debug -e dev`
> error: unknown command

When I execute `cli.ts debug --env dev`, it works fine.

If I change the code to use `-a, --env <string>` , then when I execute `cli.ts debug -a dev` it works fine.

`cli.ts`
```javascript
const program = new Command();
program.addCommand(debug);

async function main() {
  await program.parseAsync();
}
```

`debug.ts`
```javascript

const program = new Command("debug")
  .addOption(
    new Option("-e, --env <string>", "Target Environment")
      .default("dev")
      .env("ENV_NAME")
      .choices(["dev", "production"]),
  )
  .action(async (data) => {
    await debug(data as Args);
  });

// eslint-disable-next-line @typescript-eslint/no-unused-vars
const debug = async (options: Args) => {
  console.log("hi");
}
```

**Comment by shadowspawn** [comment]
1) I can't think of a change in Commander which would cause that.

2) How are you running the TypesScript file? (What is after the `#!` if you have made your TypeScript runnable?) I am wondering is a newer version of that tool is being affected by the `-e`.

3) 
The `unknown command` error from Commander has single speech marks around the command, like:
```
% fab 'x'   
error: unknown command 'x'
```

Are you seeing those in the error message? (I am wondering if the error might be coming from something else, although looks pretty similar, so just checking!)

**Comment by shadowspawn** [comment]
I have reproduced what I think is the same issue. Try changing the parse call and see if it fixes your problem. From your example code:
```
// await program.parseAsync();
await program.parseAsync(process.argv);
```

What I suspect is happening is that bun is incorrectly including script arguments into `process.execArgv` and confusing Commander's auto-detection of command-line arguments.

```js
// args.js
console.log({ argv: process.argv, execArgv: process.execArgv });
```

```console
% node args.js debug -e dev   
{
  argv: [
    '/usr/local/bin/node',
    '/Users/john/Documents/Sandpits/commander/issues/2205/args.js',
    'debug',
    '-e',
    'dev'
  ],
  execArgv: []
}
% bun run args.js debug -e dev
{
  argv: [ "/opt/homebrew/Cellar/bun/1.1.9/bin/bun", "/Users/john/Documents/Sandpits/commander/issues/2205/args.js",
    "debug", "-e", "dev"
  ],
  execArgv: [ "-e" ],
}
```

Running a program based on your example code, the script name is being parsed as a command name:
```console
% bun run cli.ts debug -e dev
error: unknown command '/Users/john/Documents/Sandpits/commander/issues/2205/cli.ts'
```

Reference: https://nodejs.org/docs/latest/api/process.html#processexecargv

**Comment by kingmesal** [comment]
Yes, that does allow it to run through the command chain.

**Comment by shadowspawn** [comment]
I have opened an issue against Bun. If it doesn't get positive activity, I will add a work-around in Commander. 

https://github.com/oven-sh/bun/issues/11673

**Comment by paperdave** [comment]
Will be fixed in https://github.com/oven-sh/bun/pull/11987, likely landing in Bun 1.1.15 or 16, within a week

**Comment by shadowspawn** [comment]
This is fixed in Bun v1.1.16, thanks to @paperclover
